parsl/tests/test_monitoring/test_memoization_representation.py

```diff
@@ -0,0 +1,81 @@
+
+import logging
+import os
+import parsl
+import pytest
+
+logger = logging.getLogger(__name__)
+
+
+
+@parsl.python_app(cache=True)
+def this_app(x):
+    return x + 1
+
+
+@pytest.mark.local
+def test_hashsum():
+    import sqlalchemy
+    from parsl.tests.configs.htex_local_alternate import fresh_config
+
+    if os.path.exists("monitoring.db"):
+        logger.info("Monitoring database already exists - deleting")
+        os.remove("monitoring.db")
+
+    logger.info("loading parsl")
+    parsl.load(fresh_config())
+
+    logger.info("invoking and waiting for result (1/4)")
+    f1 = this_app(4)
+    assert f1.result() == 5
+
+    logger.info("invoking and waiting for result (2/4)")
+    f2 = this_app(17)
+    assert f2.result() == 18
+
+    logger.info("invoking and waiting for result (3/4)")
+    f3 = this_app(4)
+    assert f3.result() == 5
+
+    logger.info("invoking and waiting for result (4/4)")
+    f4 = this_app(4)
+    assert f4.result() == 5
```

Comment:
I don't mind adding extra invocation. But I wonder about the purpose to invoke one more time here. If it is necessary, maybe another invocation with `17` would be better?

Reply:
It is to give some more distinction in the database between the this_app(4) tasks and the this_app(17) tasks, which made debugging things a bit easier for me.